### ISSUE TITLE
Questionable CS8602 'Possible dereference of null reference' with array indexer after check

### ISSUE THREAD

**Issue by Tragetaschen** [dotnet/roslyn#33278]
**Version Used**: 
16.0.0 Preview 2.2

**Steps to Reproduce**:
```csharp
void M()
{
    string?[] t = new[] { "Hello" };
    if (t[0] != null)
        t[0].Substring(1); // CS8602
}
```

**Expected Behavior**:
Right after checking `t[0]` for `null`, accessing it should not show a nullability warning.

**Actual Behavior**:
CS8602

A comparatively easy fix is to "Introduce local" => "For all occurrences"  on `t[0]`. The local variable's nullability then flows correctly.

**Comment by jaredpar** [comment]
Duplicate of #34644
